### PR TITLE
Implement client-side device web authentication

### PR DESCRIPTION
**Pull request by codingllama**
This is the basis for the [TerminalService.AuthenticateDeviceWeb RPC][1], implemented under lib/devicetrust for consistency and clean separation of responsibility. (The lib/devicetrust package is small and client-focused.)

In real-world use the DeviceWebToken is passed from the Web UI, during user login, to Connect.

https://github.com/gravitational/teleport.e/issues/3236

[1]: https://github.com/gravitational/teleport/blob/a0068878dea1b924bbe7c7259891e2892f13cfd2/proto/teleport/lib/teleterm/v1/service.proto#L174

### CODE REVIEW

File: lib/devicetrust/authn/authn.go [hunk 1]
```diff
@@ -60,14 +60,56 @@ func NewCeremony() *Ceremony {
...
+	ctx context.Context,
+	devicesClient devicepb.DeviceTrustServiceClient,
+	webToken *devicepb.DeviceWebToken,
+) error {
+	switch {
+	case devicesClient == nil:
+		return trace.BadParameter("devicesClient required")
+	case webToken == nil:
+		return trace.BadParameter("webToken required")
+	}
```
Comment (probakowski) [inline lib/devicetrust/authn/authn.go:100]: I'd rather use 2 ifs here

File: lib/devicetrust/testenv/fake_device_service.go [hunk 2]
```diff
@@ -458,17 +504,46 @@ func (s *FakeDeviceService) AuthenticateDevice(stream devicepb.DeviceTrustServic
...
+		// Spend token regardless of outcome.
+		s.deviceWebTokens = slices.Delete(s.deviceWebTokens, i, i+1)
+
+		// Validate token info.
+		switch {
+		case token.token != webToken.Token:
+			return trace.AccessDenied(invalidWebTokenMessage)
+		case token.expectedDeviceID != dev.pb.Id:
+			return trace.AccessDenied(invalidWebTokenMessage)
+		}
```
Comment (probakowski) [inline lib/devicetrust/testenv/fake_device_service.go:538]: Again, plain ifs might be easier to understand

**Review by probakowski — APPROVED** [review]

File: lib/devicetrust/authn/authn.go (hunk 1 above)
Comment (codingllama) [reply lib/devicetrust/authn/authn.go:100]: Ack. Stylistically this is fine in Go; some prefer if/else chains, some prefer switches.

File: lib/devicetrust/testenv/fake_device_service.go (hunk 2 above)
Comment (codingllama) [reply lib/devicetrust/testenv/fake_device_service.go:538]: Ack  👍

**Review by marcoandredinis — APPROVED** [review]

**Pull request merged**